crates/tabby-common/src/registry.rs

```diff
@@ -69,29 +98,61 @@ impl ModelRegistry {
         }
     }
 
-    fn get_model_dir(&self, name: &str) -> PathBuf {
+    // get_model_store_dir returns {root}/{name}/ggml, e.g.. ~/.tabby/models/TabbyML/StarCoder-1B/ggml
+    pub fn get_model_store_dir(&self, name: &str) -> PathBuf {
+        self.get_model_dir(name).join("ggml")
+    }
+
+    // get_model_dir returns {root}/{name}, e.g. ~/.tabby/models/TabbyML/StarCoder-1B
+    pub fn get_model_dir(&self, name: &str) -> PathBuf {
         models_dir().join(&self.name).join(name)
     }
 
-    pub fn migrate_model_path(&self, name: &str) -> Result<(), std::io::Error> {
-        let model_path = self.get_model_path(name);
+    // get_model_path returns the entrypoint of the model,
+    // will look for the file with the prefix "00001-of-"
+    pub fn get_model_entry_path(&self, name: &str) -> Option<PathBuf> {
```

Comment:
does this function need to be public? please also check newly added `pub` function in this struct.

Reply:
this is also used in llama-cpp-server 
https://github.com/TabbyML/tabby/pull/3258/files#diff-7e6c7f751bfa100a469d7b7e1b68d567287a88df6425e2c8155333a229ecce59R282